File: mac/Voice/Refine/Budget.swift
enum Budget {
    /// docs/PLAN.md §3.5: clamp(2500 + 12 × rawChars, 3500, 15000)
    static func ms(rawChars: Int) -> Int { min(15000, max(3500, 2500 + 12 * rawChars)) }
}

File: windows/Spit.Core/Refine/Budget.cs
namespace Spit.Core;

/// How long the client waits for `/v1/refine`. Port of mac/Voice/Refine/Budget.swift.
public static class Budget
{
    public const int MinMs = 3500;
    public const int MaxMs = 15000;

    /// docs/PLAN.md §3.5: clamp(2500 + 12 × rawChars, 3500, 15000). `rawChars` counts what Swift's
    /// `String.count` counts — grapheme clusters — so both clients give one transcript one budget.
    public static int Ms(int rawChars) => (int)Math.Clamp(2500L + 12L * rawChars, MinMs, MaxMs);
}
